packages/components/Callout/macos/Callout.swift

```diff
@@ -0,0 +1,310 @@
+		let anchorScreenRect = anchorRect.equalTo(.null) ? calculateAnchorViewScreenRect() : calculateAnchorRectScreenRect()
+		let calloutScreenRect = bestCalloutRect(relativeTo: anchorScreenRect)
+
+		calloutWindow.setFrame(calloutScreenRect, display: false)
+	}
+
+	/// Calculates the NSRect of the Anchor Rect in screen coordinates
+	private func calculateAnchorRectScreenRect() -> NSRect {
+		guard let window = window  else {
+			preconditionFailure("No window found")
+		}
+		
+		guard let screenFrame = window.screen?.visibleFrame ?? NSScreen.main?.visibleFrame else {
+			preconditionFailure("No screen Available")
+		}
+
+		// The anchor Rect is given in the coordinate space of the root view.
+		// Find the root view to convert to screen coordinates
+		var rootView: NSView = self
+		while (!rootView.isReactRootView()) {
+			rootView = rootView.reactSuperview()
+		}
+		let rootViewBoundsInWindow = rootView.convert(rootView.bounds, to: nil)
+		let rootViewRectInScreenCoordinates = window.convertToScreen(rootViewBoundsInWindow)
+		
+		// macOS uses a flipped Y coordinate (I.E: (0,0) is on the bottom left of the screen). However,
+		// React Native assumes a standard Y coordinate. Let's flip the Y coordinate of our rect to match
+		let anchorScreenRectOrigin = NSPoint(
+			x: rootViewRectInScreenCoordinates.origin.x + self.anchorRect.origin.x,
+			y: screenFrame.height - (rootViewRectInScreenCoordinates.origin.y + self.anchorRect.origin.y)
+		)
+		let anchorRectInScreenCoordinates = NSRect(origin: anchorScreenRectOrigin, size: anchorRect.size)
+		
+		return anchorRectInScreenCoordinates
+	}
+
+	/// Calculates the NSRect of the anchorView in the coordinate space of the current screen
+	private func calculateAnchorViewScreenRect() -> NSRect {
+		guard let anchorView = anchorView else {
+			preconditionFailure("No anchor view provided to position the Callout")
+		}
+
+		guard let window = window  else {
+			preconditionFailure("No window found")
+		}
+
+		let anchorBoundsInWindow = anchorView.convert(anchorView.bounds, to: nil)
+		let anchorFrameInScreenCoordinates = window.convertToScreen(anchorBoundsInWindow)
+
+		return anchorFrameInScreenCoordinates
+	}
+
+	/// Calculates the rect in screen coordinates the callout should be positioned in relative to the anchor rect, adjusting if we are close to a screen edge
+	private func bestCalloutRect(relativeTo anchorScreenRect: NSRect) -> NSRect {
+
+		guard let screenFrame = window?.screen?.visibleFrame ?? NSScreen.main?.visibleFrame else {
+			preconditionFailure("No Screen Available")
+		}
+
+		let calloutFrame = proxyView.frame
+		
+		// Find our preferred origin based on the directional hint
+		let calloutOrigin: NSPoint = {
+			var origin = NSPoint()
+
+			switch(directionalHint) {
+			case .minX:
+				origin.x = NSMinX(anchorScreenRect) - calloutFrame.size.width
+				origin.y = NSMaxY(anchorScreenRect) - calloutFrame.size.height
+				break
+			case .minY:
+				// TODO is this right?
```

Comment:
Probably shouldn't have a floating TODO in code (yes I see the irony as I probably added this in the original copy of this menu code). Also why no break in this case? Wouldn't the .maxX case just overwrite it anyway?

Reply:
Ah, good catch. FWIW I tested this and this should be right. It's not in the original copy, the original copy just asserts we don't support minY.

As it turns out, Swift switch statements don't fall through, so we don't need breaks anywhere. I should actually remove them from the other statements